main.go

```diff
@@ -85,6 +85,21 @@ type Tag struct {
 	SHA     string
 }
 
+func checkForRepCreationRequestBuilder(repo string, branch string, number int, token string) *http.Request {
```

Comment:
This function can potentially return ```nil``` but I do not see any nil checks in the code that calls this (e.g. when appending to the cowpoke requests array or in ```doRequest```)